### ISSUE TITLE
Custom Trainingset On New Usecase - Poor Results

### ISSUE THREAD

**Issue by jgcbrouns** [edvardHua/PoseEstimationForMobile#93]
Hi everyone!

First of all, thank you @edvardHua and @tucan9389  for your amazing work in this project!

I am repurposing the pose estimation for another usecase: 3D objects.
For this, I created a synthetic dataset of an Airfryer using a 3D CAD model. My trainingset consists of 6300 trainingsimages and 700 validation images. I use random images as backgrounds to simulate reflections in my 3D object. A few handpicked examples:
![airfryer_handpicked](https://user-images.githubusercontent.com/8687567/60658840-5448a800-9e54-11e9-9c61-45d3ec469ad8.jpg)

I trained a CPM model for about **113000 step**s, converted the model into .pb, then into .mlmodel and tested in on @tucan9389 IOS app and another IOS app that I found..
Using benchmark,py I get: **PCKh=99.47** I dont really understand this measurement that well, but I suppose that high values are better for PCKh.

However, performance in the app on the real-life object and the synthetic trainings-data directly is poor: [See video](https://youtu.be/MNKwqaoJHFE)

What could be the cause of this? Some thoughts I had:
- It could have something to do with me not labeling occluded points as such for training the model. I label all points as visible even if they are occluded (on the back of the object etc)
- The key-points could be inaccurate: the network-input is only 192px, which could be) too small for key-points at, for example, the buttons of the airfryer.
- The airfryer object is too hard for a convolutional model to accurately be recognized: its black, very reflective and has only few features. I personally think that this is not the case though
- There is a code-mistake somewhere
- Projection-matrix (intrinisic camera calibration etc) is incorrect
- The model itself is not accurate enough:
  - No generalization from synthetic data to real data? This is for sure something that needs to be done: domain randomization. Yet I expect the model to perform well enough on the synthetic data at least.
  - Not enough training data? I only use 6300 images for training whereas the `ai_challenger` dataset has >20.000 training images.
- I did not train for enough steps: I had to cancel the training so I reached 113000 instead of the recommended amount of steps:  num_train_samples * max_epoch' / batchsize * gpus = 6300 * 1000 / 8 = 787.500‬

However, my predictions seem to doing well in Tensorboard:
![individualImage (1)](https://user-images.githubusercontent.com/8687567/60659763-79d6b100-9e56-11e9-844e-d723b84e33ee.png)
![individualImage (2)](https://user-images.githubusercontent.com/8687567/60659764-79d6b100-9e56-11e9-80af-bd04ef5ed913.png)

And the model seems to converge according to my loss:
```
2019-07-04 09:59:45.240315: step 112970, loss = 105.70, last_heat_loss = 15.62 (42.6 examples/sec; 0.188 sec/batch)
2019-07-04 09:59:51.029643: step 112980, loss = 123.71, last_heat_loss = 17.12 (54.1 examples/sec; 0.148 sec/batch)
2019-07-04 09:59:57.532166: step 112990, loss = 96.85, last_heat_loss = 14.31 (57.2 examples/sec; 0.140 sec/batch)
2019-07-04 10:00:03.783458: step 113000, loss = 113.13, last_heat_loss = 15.40 (51.3 examples/sec; 0.156 sec/batch)
2019-07-04 10:00:26.010103: step 113010, loss = 115.02, last_heat_loss = 16.47 (55.2 examples/sec; 0.145 sec/batch)
2019-07-04 10:20:58.271013: step 113020, loss = 92.60, last_heat_loss = 13.03 (53.3 examples/sec; 0.150 sec/batch)
```

**Comment by edvardHua** [comment]
I guess the reason is training dataset is synthetic.

Through the youtube video, i noticed that the real Airfryer is quiet different to training dataset.

And i think you can label the Airfryer in real environment and train again.

Hope it helps. :)

**Comment by jgcbrouns** [comment]
hi @edvardHua 
Thanks for your respone!
And what about the amount of trainingsdata? And the amount of steps I should train?

I it possible to finetune a model that was trained, e.g: I train first on synthetic data in a first round and then use the checkpoint to train again in a second round on real data? Is this what the `checkpoint` parameter is used for in the mv2_cpm.cfg file? I can set this to a checkpoint file and continue training from there with different data?

**Comment by jgcbrouns** [comment]
I solved my issues by adding Domain Randomization. This makes the model more robust since the real version of my object becomes a 'variant' of what it has seen in training.
